torch/_dynamo/variables/base.py

```diff
@@ -185,9 +185,55 @@ def const_getattr(self, tx, name: str) -> Any:
         """getattr(self, name) returning a python constant"""
         raise NotImplementedError()
 
+    def dynamic_getattr(self, tx, name):
```

Comment:
I don't think this should go on the base class.  Applying this to every single type seems like it could have unintended consequences.

Reply:
Hmm, it passes tests and our CI suite, so I figured it was safe. Open to feedback though - is there somewhere else you'd like to see it moved to? Perhaps Tensors only?